### ISSUE TITLE
Prevent accidental Rocky to Rocky migration

### ISSUE THREAD

**Issue by bcomisky** [rocky-linux/rocky-tools#57]
I had a error running migrate2rocky due to a RPM conflict I needed to resolve before distro-sync would run.  After fixing the RPM issue I naively ran the migrate2rocky script again.  But since my system was already considered Rocky at that point, the script got confused and ended up deleting some RPMs I needed before it failed.

Log from first attempt, failed distro-sync due to RPM conflict:   https://pastebin.com/hg1KSbRL

Log from second attempt, with Rocky->Rocky failure:   https://pastebin.com/mxsi1PVf

**Comment by pajamian** [comment]
The first attempt looks to me like it would have failed a dnf update if one had been attempted before the migration.  I've seen a few cases of issues where dnf update would have failed and my determination is that if you can't successfully run dnf update then it's an indication that the distro-sync stage of the migration will also fail.  So the solution here is to have the script attempt to run dnf update before it starts the migration which will give it a chance to bail before it does anything that could leave the system in an unstable state.

The second attempt is, as you say, because you tried to run migrate2rocky a second time after the first attept was partially successful.  This can easily be solved by checking the os-release file to see if the current system is already rockylinux and bailing if it is.  On a side note here a second attempt could mean that another distro-sync and following commands is being requested, and as such we may want to have some sort of option to skip the initial system package swap and just do the distro-sync and efi commands.

**Comment by bcomisky** [comment]
Yes, you are exactly right on all counts.  I imagine I'm not the only one who will impulsively pull the trigger on migrate before doing their due diligence, so some safeguards would surely help someone else.   For reference here, a manual download of the missing rocky-* RPMS, followed by the distro-sync and fix_efi steps from the migrate2rocky script fixed my installation.

**Comment by pajamian** [comment]
Just something else I thought of, in addition to that an option that basically does the manual install of the rpms, distro sync and fix_efi steps something like, "fix failed migration" would help for those who do end up, for whatever reason, with a partial migration.

**Comment by pajamian** [comment]
> The first attempt looks to me like it would have failed a dnf update ... So the solution here is to have the script attempt to run dnf update before it starts the migration

This is done now with PR#58

**Comment by pajamian** [comment]
> The second attempt is, as you say, because you tried to run migrate2rocky a second time after the first attept was partially successful. This can easily be solved by checking the os-release file to see if the current system is already rockylinux and bailing if it is.

Done with PR #64

**Comment by systemcrash** [comment]
Allowing the script to pick up where it left off would be good... like it drops markers in a .dot file to say which steps it completed.

**Comment by pajamian** [comment]
See https://github.com/rocky-linux/rocky-tools/issues/69#issuecomment-876450537

**Comment by pajamian** [comment]
Wherever you got libnice from it wasn't CentOS, then, the latest version in CentOS 8 is 0.1.14.  Where did you get libnice from?

**Comment by lbdroid** [comment]
Here: https://koji.fedoraproject.org/koji/buildinfo?buildID=1657236
But the source of it really isn't the point. The point is that this situation broke the update. It could be any package that depends on a higher version of some package that isn't available from the repositories.
